### PR TITLE
Clean up TaskConfig definition

### PR DESCRIPTION
**Pull request by cjpatton**
- Wrie `Dp` (short for diff. privacy) instead of `Privacy`. That way we don't end up conflating this (precise) definition of privacy with other definitions. (For example, in the DAP draft, "privacy" refers to an MPC-style definition.)

- Write `Config` instead of `Params`, for consitency with `TaskConfig`, `QueryConfig`, and `VdafConfig`.

- Lift `DpConfig` out of `VdafConfig` as these apply to two separable primitives that we are composing in this application.

- Various editorial nits.

### CODE REVIEW

File: draft-wang-ppm-dap-taskprov.md [hunk 1]
```diff
@@ -162,40 +179,19 @@ struct {
...
 
 ~~~
-/* Codepoint for each standardized VDAF. Defined in
-I-D.draft-irtf-cfrg-vdaf-03. */
 enum {
-    prio3-aes128-count(0x00000000),
-    prio3-aes128-sum(0x00000001),
-    prio3-aes128-histogram(0x00000002),
-    poplar1-aes128(0x00001000),
+    prio3-aes128-count(0x00000000),     // I-D.draft-irtf-cfrg-vdaf-03
```
Comment (wangshan) [inline draft-wang-ppm-dap-taskprov.md:187]: nit: why prefer inlined comment?

File: draft-wang-ppm-dap-taskprov.md [hunk 2]
```diff
@@ -204,31 +200,32 @@ struct {
...
-when combined with privacy related parameters in `TaskConfig` like
-`min_batch_size`, can determine the VDAF's privacy guarantee.
-
 > OPEN ISSUE: Should spell out definition of `PrivacyParams` for various
 > differential privacy mechanisms and parameters.
 > See issue [#94](https://github.com/cfrg/draft-irtf-cfrg-vdaf/issues/94) for
 > discussion.
 
-The definition of Time, Duration, Url, QueryType follow those in
+The definition of `Time`, `Duration`, `Url`, QueryType follow those in
```
Comment (wangshan) [inline draft-wang-ppm-dap-taskprov.md:228]: you missed `QueryType`

**Review by wangshan — COMMENTED** [review]
Left some minor comments, mostly about missed renaming & reformatting.

@cjpatton Why bringing DpConfig to TaskConfig, given that the exact mechanism is likely to be tied to a VDAF? DpConfig is not in the core protocol's out-of-band task configuration, if people wants to introduce a new Dp config and use it out-of-band, then it'll have to be aligned with the struct defined here.

File: draft-wang-ppm-dap-taskprov.md (hunk 1 above)
Comment (cjpatton) [reply draft-wang-ppm-dap-taskprov.md]: It's easier to read, I think.

**Comment by cjpatton** [comment]
> @cjpatton Why bringing DpConfig to TaskConfig, given that the exact mechanism is likely to be tied to a VDAF?

Fair point. I guess I was a bit confused about the requirements: If the expectation is that how we use DP is *always* VDAF-dependent, then this is a fair move; but there might be ways of applying central DP that apply to a large class of VDAFs (see https://github.com/cfrg/draft-irtf-cfrg-vdaf/issues/94). 

I've merged this back into vdaf_config. If at some point we want to lift this to task_config, we can do so in a way that's backwards compatible, since dp_config comprises the first bytes of vdaf_config.

> DpConfig is not in the core protocol's out-of-band task configuration, if people wants to introduce a new Dp config and use it out-of-band, then it'll have to be aligned with the struct defined here.

Good catch!

**Review by wangshan — APPROVED** [review]

**Pull request merged**